controllers/metal3.io/preprovisioningimage_controller.go

```diff
@@ -90,56 +117,104 @@ func (r *PreprovisioningImageReconciler) Reconcile(ctx context.Context, req ctrl
 	return result, err
 }
 
+func configChanged(img *metal3.PreprovisioningImage, format metal3.ImageFormat, networkDataStatus metal3.SecretStatus) bool {
```

Comment:
This function might be a method of `PreprovisioningImage`?

Reply:
Adding methods in the API module is discouraged. This one would be harmless enough (requires no dependencies), but it isn't actually likely to be used anywhere else so there's no reason to bend the rules.